### PR TITLE
Inhomogeneous boundary conditions

### PR DESCRIPTION
**Pull request by stonecoldhughes**

### CODE REVIEW

**Comment by dlg0** [comment]
Can you add this to the `make_all_gold.m` also?

**Comment by stonecoldhughes** [comment]
sure, I can also fix that small bug

File: boundary_condition_vector.m [hunk 1]
```diff
@@ -144,7 +144,8 @@
                 fListL = bcL{d1};
                 fListR = bcR{d1};
                 
-                bcVec = bcVec + combine_dimensions_D(pde.deg,fListL,timeFacL,hash_table,opts.use_oldhash);
+                partial = combine_dimensions_D(pde.deg,fListL,timeFacL,hash_table,opts.use_oldhash);
```
Comment (dlg0) [inline boundary_condition_vector.m]: What's this `partial` variable been introduced for?

I'm unsure why we are modifying this file at all for a gold data PR?

File: combine_dimensions_D.m [hunk 2]
```diff
@@ -48,6 +48,8 @@
     tmp = A * B;
     
     Index = deg^num_dims*(i-1)+1:deg^num_dims*i;
+    % Captain! different parts of fval set on each iteration with no overlap. The same time
```
Comment (dlg0) [inline combine_dimensions_D.m]: I don't understand the "different parts of fval" comment. Are you suggesting an optimization?

As for the time multiplier, we'd discussed this some, and there may be cases where each spatial dimension function has a different time multiplier, hence the way it is done here.

File: gold/tmp_coeff.m [hunk 3]
```diff
@@ -0,0 +1,42 @@
+generate_data( diffusion1, 'diffusion1', 'lev', 2, 'deg', 2, 'grid_type', 'FG' );
```
Comment (dlg0) [inline gold/tmp_coeff.m]: what are these `tmp_` files for?

File: md_eval_function.m [hunk 4]
```diff
@@ -16,7 +16,12 @@
             md_funcs{s}{d},params,t);
     end
     
+    % Captain! This time multiplier can just be factored out. You factor out time multiplier
```
Comment (dlg0) [inline md_eval_function.m]: possibly, but I'm not sure this is true in the general case

File: md_eval_function.m [hunk 5]
```diff
@@ -16,7 +16,12 @@
...
+    % from inside combine_dimensions and apply the next two lines as axpys later.
     time_multiplier = md_funcs{s}{num_dimensions+1}(t);
+    %Captain! All these functions use the same time multiplier
+    %Captain! All combine dimensions outputs are stored in a vector and scaled by the time
```
Comment (dlg0) [inline md_eval_function.m]: yes, there is a time multiplier not related to a particular dimension, but each dimension also has its own time multiplier

File: time_advance.m [hunk 6]
```diff
@@ -66,6 +66,7 @@
     %     k3 = invMatLHS * (k3 + rhs3);
     k3 = ALHS \ (k3 + rhs3);
 else
+     % Captain! take off boundary condition vectors as a test:
```
Comment (dlg0) [inline time_advance.m]: what?

**Review by dlg0 — CHANGES_REQUESTED** [review]

File: boundary_condition_vector.m (hunk 1 above)
Comment (stonecoldhughes) [reply boundary_condition_vector.m]: this was simply so I could examine the result of the combine_dimensions_D call as I was developing

File: combine_dimensions_D.m (hunk 2 above)
Comment (stonecoldhughes) [reply combine_dimensions_D.m]: This comment was an annotation meant for me to help conceptualize what was happening - not something that would be present in the final version. I believe I was making note for myself that loop iterations did not carry data dependencies into later loop iterations.

File: gold/tmp_coeff.m (hunk 3 above)
Comment (stonecoldhughes) [reply gold/tmp_coeff.m]: I wanted to add a little bit of code to a file that already had a lot of stuff in it - so I put the new stuff in its own file while developing so that I didn't have to run the whole thing

File: md_eval_function.m (hunk 4 above)
Comment (stonecoldhughes) [reply md_eval_function.m]: Asgard is designed as if that statement is true, so I just followed the pattern.

File: md_eval_function.m (hunk 5 above)
Comment (stonecoldhughes) [reply md_eval_function.m]: these are more semi-coherent annotations I put in the code for myself as I was figuring out how it worked

File: time_advance.m (hunk 6 above)
Comment (stonecoldhughes) [reply time_advance.m]: A semi coherent narration of some test I was doing. I think I was checking to see if results agreed before incorporating the boundary conditions

**Comment by dlg0** [comment]
@stonecoldhughes  What's the deal with this PR?

**Comment by bmcdanie** [comment]
changes updated and pulled into #47 - don't need this PR.
